core/src/main/java/org/apache/iceberg/actions/BaseSnapshotTableActionResult.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+public class BaseSnapshotTableActionResult implements SnapshotTable.Result {
+
+  private final long importedDataFilesCount;
+
+  public BaseSnapshotTableActionResult(long importedDataFilesCount) {
+    this.importedDataFilesCount = importedDataFilesCount;
+  }
+
+  @Override
+  public long importedDataFilesCount() {
+    return importedDataFilesCount;
```

Comment:
Not for this pr, but now that I think about it we should probably also let the user know how many metadata files were created as well

Reply:
Good idea! Now that we have the result interface, we can evolve it. Could you create an issue for this, @RussellSpitzer? 

@flyrain, would you be interested to pick it up?

Reply:
#2439